SwanHelp/src/index.ts

```diff
@@ -1,25 +1,44 @@
 import { JupyterFrontEnd, JupyterFrontEndPlugin } from '@jupyterlab/application';
 import { ConfigWithDefaults, ConfigSection } from '@jupyterlab/services';
-import { IFrame, MainAreaWidget } from '@jupyterlab/apputils';
+import { Dialog, IFrame, MainAreaWidget, ICommandPalette } from '@jupyterlab/apputils';
 import { IMainMenu } from '@jupyterlab/mainmenu';
-import swanDialog from './dialog';
+import { ILauncher } from '@jupyterlab/launcher';
+import { showAboutDialog, showLoadingSpinnerDialog } from './dialog';
+import { swanGalleryIcon } from './icons';
+import { downloadUrlFromServer } from './gallery-download-helper';
+import { showDialog } from "@jupyterlab/apputils";
+
+function registerGalleryListener(app: JupyterFrontEnd, galleryUrl: string) {
+  window.addEventListener('message', async (event: any) => {
+    if (event.origin != galleryUrl) {
+      console.error(`SwanGallery: Failed to validate origin, message from ${event.origin} does not match ${galleryUrl} . Skipping downloading URL`);
+      return;
+    }
+    const dialog = showLoadingSpinnerDialog();
+    try {
+      const response = await downloadUrlFromServer(event.data);
+
+      app.commands.execute('filebrowser:open-path', {
+        path: response.path,
+        showBrowser: false
+      });
+      dialog.reject();
+    } catch (error) {
+      dialog.reject();
+      showDialog({ title: "Failed to download notebook", buttons: [Dialog.okButton()] });
+      console.error(
```

Comment:
This error message is a bit strange. We might have the extension running and fail for smth else. But this is not necessarily user visible, but I would still put some generic message.